site/pages/api/create-disruption.api.ts

```diff
@@ -1,119 +1,44 @@
-
-    const summary = formFields.summary;
-
-    validateSummary(summary, errors, "some-error-id");
-
-    const description = formFields.description;
-
-    validateDescription(description, errors, "some-error-id");
-
-    const associatedLink = formFields["associated-link"];
-
-    if (associatedLink) {
-        validateAssociatedLink(associatedLink, errors, "some-error-id");
-    }
-
-    const disruptionReason: string = formFields["disruption-reason"] || "";
-
-    validateDisruptionReasons(disruptionReason, errors, "some-error-id");
-
-    const disruptionStartDate = formFields["disruption-start-date"];
-    const disruptionStartTime = formFields["disruption-start-time"];
-    const disruptionEndDate = formFields["disruption-end-date"];
-    const disruptionEndTime = formFields["disruption-end-time"];
-    const disruptionIsNoEndDateTime = formFields["disruption-no-end-date-time"];
-
-    validateDateTimeSection(
-        disruptionStartDate,
-        disruptionStartTime,
-        disruptionEndDate,
-        disruptionEndTime,
-        disruptionIsNoEndDateTime,
-        errors,
-        "some-error-id",
-    );
-    const publishStartDate = formFields["publish-start-date"];
-    const publishStartTime = formFields["publish-start-time"];
-    const publishEndDate = formFields["publish-end-date"];
-    const publishEndTime = formFields["publish-end-time"];
-    const publishIsNoEndDateTime = formFields["publish-no-end-date-time"];
-
-    validateDateTimeSection(
-        publishStartDate,
-        publishStartTime,
-        publishEndDate,
-        publishEndTime,
-        publishIsNoEndDateTime,
-        errors,
-        "some-error-id",
-    );
-
-    const disruptionData: DisruptionPageInputs = {
-        "type-of-disruption": disruptionType,
-        summary: summary,
-        description: description,
-        "associated-link": associatedLink,
-        "disruption-reason": disruptionReason,
-        "disruption-start-date": disruptionStartDate
-            ? getDateTime(disruptionStartDate).toString()
-            : disruptionStartDate,
-        "disruption-end-date": disruptionEndDate ? getDateTime(disruptionEndDate).toString() : disruptionEndDate,
-        "disruption-start-time": disruptionStartTime,
-        "disruption-end-time": disruptionEndTime,
-        "disruption-no-end-date-time": disruptionIsNoEndDateTime,
-        "publish-start-date": publishStartDate ? getDateTime(publishStartDate).toString() : publishStartDate,
-        "publish-end-date": publishEndDate ? getDateTime(publishEndDate).toString() : publishEndDate,
-        "publish-start-time": publishStartTime,
-        "publish-end-time": publishEndTime,
-        "publish-no-end-date-time": publishIsNoEndDateTime,
-        "disruption-repeats": formFields["disruption-repeats"],
-    };
-
-    setCookieOnResponseObject(COOKIES_DISRUPTION_INFO, JSON.stringify(disruptionData), res);
+    try {
+        const validatedBody = createDisruptionsSchemaRefined.parse(req.body);
```

Comment:
Is there something we gain from using .parse and catching the error as opposed to using safeParse and checking the result? Something like:

const validationResult = createDisruptionsSchemaRefined.parse(req.body);

if (!validationResult.success) {
 <failure case code>
} else {
<success case code>
}

And then we can wrap the whole thing in a try catch (similar to what's done on CFDS API's) as per the norm.

There might be something I'm not aware of which means we want to use the normal parse, just thought I'd ask.